readers/influxdb/messages_test.go

```diff
@@ -58,31 +60,37 @@ func TestReadAll(t *testing.T) {
 	writer := writer.New(client, testDB)
 
 	messages := []senml.Message{}
-	subtopicMsgs := []senml.Message{}
+	valSubtopicMsgs := []senml.Message{}
+	boolMsgs := []senml.Message{}
+	stringMsgs := []senml.Message{}
+	dataMsgs := []senml.Message{}
+
 	now := time.Now().UnixNano()
 	for i := 0; i < msgsNum; i++ {
 		// Mix possible values as well as value sum.
-		count := i % valueFields
 		msg := m
+		msg.Time = float64(now)/float64(1e9) - float64(i)
```

Comment:
`msg.Time = float64(now)/1e9 - i` should be enough for the same reasons specified above.

Reply:
Without to  cast this returns `mismatched types float64 and int`.  Let's keep this one

Reply:
I removed the unnecessary cast of `1e9` the one from `i` is needed because it's an `int`.

Reply:
This `i` will be auto-converted to float64 because of the precision of the left operand `float64(now)/1e9`. So the whole expression will be float64 type. If you need an expression to be a float64 type, then this should work and it seems to me that you need it.

Reply:
However, you're right about `i` because you implictely declared it as an int and i used a literal value.

Reply:
I fact it's not correct. In Go there is no implicit casting with variables. Here probably `i`is forcing to cast all values.

Reply:
In fact, there is no casting in go. Only converting. And there is implicit converting. See here https://golang.org/ref/spec#Conversions

Reply:
As far as I understand Go can implicitly convert only literal values that happen to be in expressions, so that's why this works:

```
nowF := float64(now)/1e9 - 100
```

while this does not work:

```
i := 100
nowF := float64(now)/1e9 - i
```

because Go does not implicitly converts `i` to float64, but converts litteral value of 100 to float64 (please see here https://yourbasic.org/golang/conversions/#implicit-conversions also).

Reply:
Yes,that's it.I reverted too previous version. I forgot but I already had this issue. I changed both at the same time... At least the difference is clear now.

Reply:
IMO, you should still use `iVal := int64(fVal * 1e9)`, because you fVal is float64 and 1e9 is implicitly converted to float64 per explanations above. 

The same goes for `float64(now)/1e9`. Only the `i` cannot be converted implicitly, because it's type is already determined as int. Implicit conversation works only on literal value.

Reply:
You are right. I got confused with docker tests. There is a strange behavior if you run tests several times. Sometimes it fails and if you repeat then it pass... it could be some docker tests cache? 
In any case I removed the cast, it should be ok now.